Engine/Fuego/Windows/EventQueueWin.cpp

```diff
@@ -6,24 +6,24 @@ void EventQueueWin::Update()
 {
 }
 
-std::shared_ptr<Event> EventQueueWin::Front()
+std::unique_ptr<EventVariant>& EventQueueWin::Front()
```

Comment:
I can't recomend you to use unique_ptr referencves, epecially non const: 
1. It breaks down semantics of uniqueness of resource ownership.
2. What happen if recieving code call `std::move` or `reset`? If you afraid of cyclic references - use `weak_ptr` and `shared_ptr`.

Reply:
Changed back to UPtr